runtime-modules/council/src/lib.rs

```diff
@@ -527,6 +607,68 @@ impl<T: Trait> Module<T> {
         Self::deposit_event(RawEvent::AnnouncingPeriodStarted());
     }
 
+    /////////////////// Budget-related /////////////////////////////////////
+
+    /// pay rewards to elected council members
+    fn pay_elected_member_rewards(now: T::BlockNumber) {
+        let reward_per_block = T::ElectedMemberRewardPerBlock::get();
+
+        // intermediate balance
+        let mut new_balance = Budget::<T>::get();
+
+        // pay reward to all council members
+        for (member_index, council_member) in CouncilMembers::<T>::get().iter().enumerate() {
+            // stop iterating if budget is completely depleted
+            if new_balance == 0.into() {
+                break;
+            }
+
+            // calculate unpaid reward
+            let unpaid_reward =
+                Calculations::<T>::get_current_reward(&council_member, reward_per_block, now);
+
+            if unpaid_reward == 0.into() {
+                continue;
+            }
+
+            // calculate withdrawable balance
+            let (available_balance, missing_balance) =
+                Calculations::<T>::payable_reward(&new_balance, &unpaid_reward);
+
+            // pay reward
+            Mutations::<T>::pay_reward(
+                member_index,
+                &council_member.staking_account_id,
+                &available_balance,
+                &missing_balance,
+                &now,
+            );
+
+            // remember new balance
+            new_balance -= available_balance;
+
+            // emit event
+            if missing_balance > 0.into() {
```

Comment:
If you were unable to pay out the current `council_member`, aren't all payouts now impossible because `new_balance == 0`? If this is true, then
a) lets make the conditional on `new_balance`.
b) I think the general iteration should stop, but we should still iterate over the remaining council members and emit events saying they did not get paid at all and what their current owed amount is. One way to do this is to just introduce a first `if` inside the loop which automatically sidesteps all the member specific calculations which are known to fail. Also see my comment on suggested event changes. Let me know if you have any cleaner suggestion here.

Reply:
An event is now emitted for all council members. Only one event, `RewardPayment`, with info about how much was paid off and how much unpaid reward exists now.